libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/Milestone.kt

```diff
@@ -0,0 +1,120 @@
+package com.mapbox.services.android.navigation.v5.milestone
+
+import com.mapbox.services.android.navigation.v5.instruction.Instruction
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+
+/**
+ * Base Milestone statement. Subclassed to provide concrete statements.
+ *
+ * @since 0.4.0
+ */
+abstract class Milestone
+// Public exposed for creation of milestone classes outside SDK
+constructor(private val builder: Builder) {
```

Comment:
Could this be a refactored into primary constructor?

Reply:
It's already primary)